examples/image_filtering/src/common/graphic.ads

```diff
@@ -0,0 +1,45 @@
+------------------------------------------------------------------------------
+--                       Copyright (C) 2017, AdaCore                        --
+-- This is free software;  you can redistribute it  and/or modify it  under --
+-- terms of the  GNU General Public License as published  by the Free Soft- --
+-- ware  Foundation;  either version 3,  or (at your option) any later ver- --
+-- sion.  This software is distributed in the hope  that it will be useful, --
+-- but WITHOUT ANY WARRANTY;  without even the implied warranty of MERCHAN- --
+-- TABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public --
+-- License for  more details.  You should have  received  a copy of the GNU --
+-- General  Public  License  distributed  with  this  software;   see  file --
+-- COPYING3.  If not, go to http://www.gnu.org/licenses for a complete copy --
+-- of the license.                                                          --
+------------------------------------------------------------------------------
+
+with Ada.Unchecked_Deallocation;
+
+package Graphic is
+   
+   type Rgb is record
+      R, G, B : Float;
```

Comment:
Use Component instead of Float as the type there

Reply:
Ok. Then what becomes my strategy when accumulating inside the kernel? eg. `Filtered_Rgb := Filtered_Rgb + (Img (X, Y) * Sg_Cdg);` goes way over 255.0.

Reply:
Ah I see, in that case it would make sense to add another float type for the accumulator and the divisor, so that we can be sure the value is ultimately back to the proper range, I’m leaving it to you to decide whether this is worth it or not